oracle/oracle_test.go

```diff
@@ -93,6 +96,133 @@ func Test_Oracle_ManualSubscription(t *testing.T) {
 
 // TODO: Mix manual and automatic subscriptions
 
+// Simulates 100 slots with "AdvanceStateToNextSlot". Each slot is configured randomly with a
+// new sub, unsub or donation. The block proposed can be okproposal, missed or wrongfee.
+// these are all randomly set each block
+func Test_100_slots_test(t *testing.T) {
+
+	//set new oracle instance
+	oracle := NewOracle(&config.Config{
+		Network:               "mainnet",
+		PoolAddress:           "0xdead000000000000000000000000000000000000",
+		UpdaterAddress:        "",
+		DeployedSlot:          uint64(50000),
+		CheckPointSizeInSlots: uint64(100),
+		PoolFeesPercent:       5,
+		PoolFeesAddress:       "0xfee0000000000000000000000000000000000000",
+		CollateralInWei:       big.NewInt(1000000),
+	})
+
+	subsIndex := make([]uint64, 0)
+	totalAssets := big.NewInt(0)
+	const seed int64 = 50000
+	rand.Seed(seed)
+	fmt.Printf("Execution SEED: %d\n", seed)
+	// main loop, iterates through 100 slots
+	for i := 0; i <= 99; i++ {
+		newSubscription := make([]Subscription, 0)
+		newUnsubscription := make([]Unsubscription, 0)
+		don := make([]Donation, 0)
+		fmt.Println("NEW BLOCK:")
```

Comment:
can we add more information on the logs instead of just NEW BLOCK? for example, log every event that you simulate.
* mocked subscription
* mocked proposal
* mocked donation
* mocked unsubscription

and in each low include some information on the event.

the idea is to be able to kind of read the logs and understand what is happening, so since we are mocking the onchain stuff (donations, subs, unsubs) its would be nice to have some logs to know whats happening (what we are feeding the oracle with)

Reply:
New commit implements info about 
-  sub
- unsub
- donation
- validator index of block proposer
- Type of block to propose (proposalOk, wrongfee or missed)

For each block, output is the following
![image](https://user-images.githubusercontent.com/36164126/231495316-0109dd18-faf9-426c-b33d-1686397d7956.png)
Interesting: By pure luck, in this block the block proposer (50005) also unsubscribes from the pool. He does a proposalOk block.